mapper/thin_edge_json/src/json.rs

```diff
@@ -354,19 +337,62 @@ mod tests {
         }"#;
         let expected_error = r#"Invalid ISO8601 timestamp (expected YYYY-MM-DDThh:mm:ss.sss.±hh:mm): "2013-06-2217:03:14.000658767+02:00": input contains invalid characters"#;
         let json = json::parse(&input).unwrap();
-        match &json {
-            JsonValue::Object(thin_edge_obj) => {
-                for (k, v) in thin_edge_obj.iter() {
-                    if k.eq("time") {
-                        let output_error =
-                            ThinEdgeJson::check_timestamp_for_iso8601_complaint(v).unwrap_err();
-                        assert_eq!(output_error.to_string(), expected_error);
-                    }
-                }
-            }
-            _ => {}
-        }
+        let time = json.entries().nth(0).unwrap();
+        let output = ThinEdgeJson::check_timestamp_for_iso8601_complaint(time.1).unwrap_err();
+        assert_eq!(output.to_string(), expected_error);
+    }
+
+    #[test]
+    fn check_from_json_str_short_timestamp() {
+        let input = r#"{
+            "temperature" : 25
+        }"#;
+        let timestamp = FixedOffset::east(2 * 3600)
+            .ymd(2021, 04, 30)
+            .and_hms_milli(17, 03, 14, 123);
+
+        //prepare expected output
+        let tvalues: ThinEdgeValue = ThinEdgeValue::Single(SingleValueMeasurement {
+            name: "temperature".into(),
+            value: 25.0,
+        });
+        let tjson: ThinEdgeJson = ThinEdgeJson {
+            timestamp,
+            values: vec![tvalues],
+        };
+        let expected_string = format!("{:#?}", tjson);
+
+        let output = ThinEdgeJson::from_str(input, timestamp).unwrap();
+        let output_string = format!("{:#?}", output);
+
+        assert_eq!(output_string, expected_string);
     }

```

Comment:
This test might test the correct point but this is not obvious.

Why not:

```
    #[test]
    fn thin_edge_json_accept_nano_second_timestamp() {
        let input = r#"{
            "time" : "2021-04-30T17:03:14.123456789+02:00",
            "temperature" : 25
        }"#;

        let output = ThinEdgeJson::from_str(input, test_timestamp()).unwrap();

        assert_eq!(output.timestamp,
                   FixedOffset::east(2 * 3600)
                    .ymd(2021, 04, 30)
                    .and_hms_nano(17, 03, 14, 123456789));
    }
```

Reply:
I was trying to compare whole input message, instead of just timestamp. Anyways, I can change it.

Reply:
Why ignore this example?

Reply:
I dont know some how I missed it!. I fixed it now.